app/controllers/application_controller.rb

```diff
@@ -23,7 +23,16 @@ def current_merchant?
   def current_admin
     current_user && current_user.admin?
   end
-  
-  
-  
+
+  def require_current_user
+    render file: "/public/404" unless current_user
+  end
+
+  def require_merchant_employee
+    render file: "/public/404" unless current_merchant?
+  end
+
+  def require_admin
```

Comment:
I also have this in the admin controller right now, so we just need to pick one and stick with it